easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.5.0-goolfc-2017b-Python-3.6.3.eb

```diff
@@ -0,0 +1,38 @@
+name = 'TensorFlow'
+version = '1.5.0'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://www.tensorflow.org/'
+description = "An open-source software library for Machine Intelligence"
+
+toolchain = {'name': 'goolfc', 'version': '2017b'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/tensorflow/tensorflow/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = [
+    'TensorFlow-%(version)s_swig-env.patch',
+    'TensorFlow-%(version)s_dont_expand_cudnn_path.patch',
+    'TensorFlow-%(version)s_dont_expand_cuda_path.patch',
+]
+checksums = [
+    '0642781c3a3a8c2c4834b91b86aec385f0b2ada7d721571458079478cc5b29c8',  # v1.5.0.tar.gz
+    '53807290f1acb6a0f2a84f1a0ad9f917ee131c304b3e08679f3cbd335b22c7ef',  # TensorFlow-1.5.0_swig-env.patch
+    '93ba981296cf4e0d2411fbf6380dd325fa41ebff5d82b02ffe7747f733bca579',  # TensorFlow-1.5.0_dont_expand_cudnn_path.patch
+    '3dcefd56748a6e1bd0cfb1e08e0b33d5d47dbaed7cb706dd7e6842413faf54a7',  # TensorFlow-1.5.0_dont_expand_cuda_path.patch
+]
+
+builddependencies = [
+    ('Bazel', '0.7.0'),
+    ('wheel', '0.30.0', versionsuffix),
+]
+dependencies = [
+    ('Python', '3.6.3'),
+    ('cuDNN', '7.0.5.15'),
+]
+
+with_mkl_dnn = False
+
+cuda_compute_capabilities = ['3.0', '3.2', '3.5', '3.7']
```

Comment:
@akesandgren How did you pick this exactly? I'm not sure what the best default would be here...

Reply:
That is what is currently useful for us. The main problem is that CUDA 9 doesn't support 2.0 which is in the default list for tensorflow 1.5 and it must be removed.
One could pick up the existing list from tensorflow, remove 2.0 when CUDA is >= 9 and be done with it.
Or just set it to the current list which i think should be 3.0, 3.2, 3.5, 3.7, 5.0 6.0, 7.0 but i didn't check the actual cuda 9 list. (And 7.0 is not supported in cuda 8)

Reply:
At the very least, we should include a pointer to https://developer.nvidia.com/cuda-gpus .

As for the default value, I think it makes sense to also include 5.2, 6.0, 6.1 and 7.0 (based on the list of Tesla GPUs). Or would that cause problems when building TF on a system that doesn't support 5.x & up?

Not sure why you need 3.2?

Reply:
3.2 is for Jetson TK1

If the list is kept in this recipy then the full list excluding 2.0 can be used since it is based on CUDA 9

If done in the easyblock then it needs to be created based on CUDA version which is fairly easy, but needs to be updated every time a new CUDA comes out...

Reply:
So, should i change the list in the recipy or should we move this to the easyblock?

Reply:
It makes sense to me to keep this in the easyconfig file, since it's closely tied to the CUDA version being used.

We should just try and make sure we also target 'modern' cards by default, to get proper performance.

Here's what is used by default for the TF binaries: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/Dockerfile.devel-gpu#L88 .

See also https://github.com/tensorflow/tensorflow/issues/6914 .

Not including `7.0` is a mistake imho, I consider performance a bigger factor than code size or compilation time...